chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml

```diff
@@ -1,138 +1,189 @@
-  args:
-    executable: /bin/bash
-  register: pod_status
-
-- name: Check if the pod is Evicted
-  shell: >
-    kubectl get pod {{ pod_name.stdout }} -n {{ a_ns }} -o jsonpath='{.status.reason}'
-  args:
-    executable: /bin/bash
-  register: eviction_status
-
-- name: Delete Pod if evicted
-  shell: >
-    kubectl delete pod {{ pod_name.stdout }} -n {{ a_ns }}
-  when: eviction_status.stdout == "Evicted"
-
-- name: If Pod is not evicted / running
-  shell: >
-    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ a_ns }} -- sh -c "rm -rf /diskfill/{{ containerID.stdout }}/diskfill"
-  when: pod_status.stdout == "Running"
-
-- name: Delete DaemonSet disk-fill
-  shell: >
-    kubectl delete ds disk-fill
+    - name: Record the disk-fill pod on app node
+      shell: >
+        kubectl get pods -l app=disk-fill -o jsonpath='{.items[?(@.spec.nodeName=="{{app_node.stdout}}")].metadata.name}'
+      args:
+        executable: /bin/bash
+      register: disk_fill_pod
+
+    - name: Exec into the pod to get ephemeral-storage size
+      shell: >
+        kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ a_ns }} -- sh -c "cd /diskfill && cd {{ containerID.stdout }} && du -k |tail -n1 | awk '{print $1}'"
+      args:
+        executable: /bin/bash
+      register: pod_current_size_full
+
+    - name: Echo the filled storage in KB to get the integer part for multiplication
+      shell: >
+        echo {{ pod_current_size_full.stdout }} | grep -Eo '+|[0-9]+'
+      args:
+        executable: /bin/bash
+      register: pod_current_size
+
+    - name: Total Used Ephemeral Storage in KB
+      set_fact:
+        used_storage: "{{ pod_current_size.stdout }}"
+  
+    - name: Subtract to get the file allocation size
+      set_fact:
+        fill_storage: "{{ to_fill|int - used_storage|int }}"
+
+    - include_tasks: /chaoslib/litmus/disk_fill/file_creation.yml
+
+    - name: Waiting for Chaos Duration
+      wait_for:
+        timeout: "{{ c_duration }}"
+
+    - name: Get the Pod Status 
+      shell: >
+        kubectl get pod {{ pod_name.stdout }} -n {{ a_ns }} -o jsonpath='{.status.phase}'
+      args:
+        executable: /bin/bash
+      register: pod_status
+
+    - name: Check if the pod is Evicted
+      shell: >
+        kubectl get pod {{ pod_name.stdout }} -n {{ a_ns }} -o jsonpath='{.status.reason}'
+      args:
+        executable: /bin/bash
+      register: eviction_status
+
+    - name: Delete Pod if evicted
+      shell: >
+        kubectl delete pod {{ pod_name.stdout }} -n {{ a_ns }}
+      when: eviction_status.stdout == "Evicted"
+
+    - name: If Pod is not evicted / running
+      shell: >
+        kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ a_ns }} -- sh -c "rm -rf /diskfill/{{ containerID.stdout }}/diskfill"
+      when: pod_status.stdout == "Running"
+
+    ## DELETING THE DAEMONSET
+    - name: Deleting the Disk Fill DaemonSet
+      shell: kubectl delete ds -l app=disk-fill
```

Comment:
Can you use `kubectl delete -f /chaoslib/litmus/disk_fill/disk_fill_ds.yaml -n {{ a_ns }}`